src/item.cpp

```diff
@@ -5733,11 +5785,8 @@ int item::get_coverage( const bodypart_id &bodypart ) const
 
     int coverage = 0;
     for( const armor_portion_data &entry : t->data ) {
-        if( entry.covers.has_value()
-            && ( entry.covers->test( bodypart.id() ) || bodypart == bodypart_id( "num_bp" ) ) ) {
-            if( entry.coverage > coverage ) {
-                coverage = entry.coverage;
-            }
+        if( entry.covers.has_value() && entry.covers->test( bodypart.id() ) && entry.coverage > coverage ) {
+            coverage = entry.coverage;
         }
     }
```

Comment:
I can't remember for sure right now, but I suspect you have a few of these loops that are intending to find the `armor_portion_data` for a particular bodypart.  Suggest you factor this pattern out into a helper function, something like

```c++
cata::optional<armor_portion_data> item::portion_for_bodypart( const bodypart_id &bodypart ) const;
```

Then this code simply becomes
```c++
int item::get_coverage( const bodypart_id &bodypart ) const
{
    if( cata::optional<armor_portion_data> portion_data = portion_for_bodypart( bodypart ) ) {
        return portion_data->coverage;
    }
    return 0;
}
```